src/config/mod.rs

```diff
@@ -0,0 +1,195 @@
+use serde::{Deserialize, Serialize};
+use std::error;
+use std::io::Write;
+use uuid::Uuid;
+
+use tiny_http::{Response, Server};
+
+use crate::ha_api;
+use crate::ha_api::GetAccessTokenResponse;
+use std::collections::HashMap;
+use url::Url;
+
+use tungstenite::{connect, Message};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct HaConfig {
+    pub host: String,
+    #[serde(rename = "long-lived-token")]
+    pub long_lived_token: Option<String>,
+    #[serde(rename = "device-id")]
+    pub device_id: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct YamlConfig {
+    pub ha: HaConfig,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct WsAuthRequest {
+    #[serde(rename(serialize = "type"))]
+    auth_type: String,
+    access_token: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct WsLongLivedAccessTokenRequest {
+    id: u32,
+    #[serde(rename(serialize = "type"))]
+    command_type: String,
+    client_name: String,
+    lifespan: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct WsLongLivedAccessTokenResponse {
+    id: u32,
+    #[serde(rename(deserialize = "type"))]
+    command_type: String,
+    success: bool,
+    result: Option<String>,
+}
+
+type Result<T> = std::result::Result<T, Box<dyn error::Error>>;
+
+pub async fn wait_for_token(config: &YamlConfig) -> Result<GetAccessTokenResponse> {
+    let server = Server::http("0.0.0.0:8000").unwrap();
```

Comment:
This address appears a lot around the application, might be nice to extract it out to a const.

And use said const in the URL below.